### PR TITLE
Update nullable annotations in CodeFixes folder

### PR DESCRIPTION
**Pull request by sharwell**

### CODE REVIEW

File: src/Workspaces/Core/Portable/CodeFixes/FixAllOccurrences/BatchFixAllProvider.cs [hunk 1]
```diff
@@ -198,19 +201,10 @@ public override async Task<CodeAction> GetFixAsync(FixAllContext fixAllContext)
...
-            Debug.Assert(!diagnostics.IsDefault);
-            cancellationToken.ThrowIfCancellationRequested();
-
-            var registerCodeFix = GetRegisterCodeFixAction(fixAllState, fixes);
-            var context = new CodeFixContext(
-                project, diagnostics, registerCodeFix, cancellationToken);
-
-            // TODO: Wrap call to ComputeFixesAsync() below in IExtensionManager.PerformFunctionAsync() so that
-            // a buggy extension that throws can't bring down the host?
-            return fixAllState.CodeFixProvider.RegisterCodeFixesAsync(context) ?? Task.CompletedTask;
```
Comment (sharwell) [inline src/Workspaces/Core/Portable/CodeFixes/FixAllOccurrences/BatchFixAllProvider.cs:210]: 📝 I'm hoping this wasn't used. Normal code fixes don't support no-location diagnostics, and the SuppressMessage batch fixer overrides this method anyway.

Comment (mavasani) [reply src/Workspaces/Core/Portable/CodeFixes/FixAllOccurrences/BatchFixAllProvider.cs:210]: Yes, I think this is only used by suppression fixer when invoked for bulk suppressing diagnostics from error list, which can include project diagnostics.

File: src/Workspaces/SharedUtilitiesAndExtensions/Workspace/Core/CodeFixes/FixAllContextHelper.cs [hunk 2]
```diff
@@ -86,14 +88,15 @@ internal static class FixAllContextHelper
             var treeToDocumentMap = await GetTreeToDocumentMapAsync(projects, cancellationToken).ConfigureAwait(false);
 
             var builder = ImmutableDictionary.CreateBuilder<Document, ImmutableArray<Diagnostic>>();
-            foreach (var documentAndDiagnostics in diagnostics.GroupBy(d => GetReportedDocument(d, treeToDocumentMap)))
+            foreach (var (document, diagnosticsForDocument) in diagnostics.GroupBy(d => GetReportedDocument(d, treeToDocumentMap)))
             {
+                if (document is null)
```
Comment (mavasani) [inline src/Workspaces/SharedUtilitiesAndExtensions/Workspace/Core/CodeFixes/FixAllContextHelper.cs:93]: Is this a bug fix? Did we add unit test?

**Review by mavasani — APPROVED** [review]
LGTM. Just a small question about what seems to be a bug fix.

File: src/Workspaces/SharedUtilitiesAndExtensions/Workspace/Core/CodeFixes/FixAllContextHelper.cs (hunk 2 above)
Comment (sharwell) [reply src/Workspaces/SharedUtilitiesAndExtensions/Workspace/Core/CodeFixes/FixAllContextHelper.cs:93]: 💭 I'm not sure whether or not this case is reachable

Comment (mavasani) [reply src/Workspaces/SharedUtilitiesAndExtensions/Workspace/Core/CodeFixes/FixAllContextHelper.cs:93]: Sounds good, I think the PR is fine as-is.

**Pull request merged**